lisa/sut_orchestrator/azure/common.py

```diff
@@ -325,16 +325,19 @@ def get_environment_context(environment: Environment) -> EnvironmentContext:
     return environment.get_context(EnvironmentContext)
 
 
-def wait_operation(operation: Any) -> Any:
-    timeout = 60 * 5
-    timer = create_timer()
-    while timeout > timer.elapsed(False):
+def wait_operation(operation: Any, time_out: int = 0) -> Any:
+    if time_out:
+        timer = create_timer()
+        while time_out > timer.elapsed(False):
+            check_cancelled()
+            if operation.done():
+                break
+            operation.wait(1)
+        if time_out < timer.elapsed():
+            raise Exception("Timeout on operation")
+    else:
```

Comment:
even it's not set, it's still needed to wait by one second. It makes the check_cancelled run more frequently.